File: c#/algorithms/sort/short_bubble.cs
using System;
using System.Collections.Generic;
using System.Linq;

List<int> numbers = new List<int>();

// Generate a list of 20 random numbers
Random random = new Random();
for (int i = 0; i < 20; i++)
{
    int number = random.Next(1, 1001);
    numbers.Add(number);
}

Console.WriteLine("Original list: " + string.Join(", ", numbers));

// Short bubble sort algorithm
void ShortBubbleSort(List<int> arr)
{
    int n = arr.Count;
    bool swapped = true;
    int passes = 0;

    while (swapped)
    {
        swapped = false;
        passes++;

        for (int i = 0; i < n - passes; i++)
        {
            if (arr[i] > arr[i + 1])
            {
                // Swap the elements
                int temp = arr[i];
                arr[i] = arr[i + 1];
                arr[i + 1] = temp;
                swapped = true;
            }
        }
    }
}

// Sort the list using short bubble sort
ShortBubbleSort(numbers);

Console.WriteLine("Sorted list in ascending order: " + string.Join(", ", numbers));

File: cpp/algorithms/sort/short_bubble.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <random>

using namespace std;

void shortBubbleSort(vector<int> &arr)
{
    int n = arr.size();
    bool swapped = true;
    int passes = 0;

    while (swapped)
    {
        swapped = false;
        passes++;

        for (int i = 0; i < n - passes; i++)
        {
            if (arr[i] > arr[i + 1])
            {
                swap(arr[i], arr[i + 1]);
                swapped = true;
            }
        }
    }
}

int main()
{
    // Generate a list of 20 random numbers
    vector<int> numbers;
    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> dis(1, 1000);

    for (int i = 0; i < 20; i++)
    {
        numbers.push_back(dis(gen));
    }

    cout << "Original list: ";
    for (int i = 0; i < numbers.size(); i++)
    {
        cout << numbers[i] << " ";
    }
    cout << endl;

    // Sort the list using short bubble sort
    shortBubbleSort(numbers);

    cout << "Sorted list in ascending order: ";
    for (int i = 0; i < numbers.size(); i++)
    {
        cout << numbers[i] << " ";
    }
    cout << endl;

    return 0;
}
